x-pack/plugins/security_solution/public/management/pages/trusted_apps/view/trusted_apps_page.test.tsx

```diff
@@ -9,22 +9,79 @@ import { TrustedAppsPage } from './trusted_apps_page';
 import { AppContextTestRender, createAppRootMockRenderer } from '../../../../common/mock/endpoint';
 import { fireEvent } from '@testing-library/dom';
 import { MiddlewareActionSpyHelper } from '../../../../common/store/test_utils';
-import { NewTrustedApp, PostTrustedAppCreateResponse } from '../../../../../common/endpoint/types';
+import {
+  ConditionEntryField,
+  GetTrustedListAppsResponse,
+  NewTrustedApp,
+  OperatingSystem,
+  PostTrustedAppCreateResponse,
+  TrustedApp,
+} from '../../../../../common/endpoint/types';
 import { HttpFetchOptions } from 'kibana/public';
+import { TRUSTED_APPS_LIST_API } from '../../../../../common/endpoint/constants';
 
 jest.mock('@elastic/eui/lib/services/accessibility/html_id_generator', () => ({
   htmlIdGenerator: () => () => 'mockId',
 }));
 
+//
+//
+//
+jest.setTimeout(30000);
+//
+//
+//
+
 describe('When on the Trusted Apps Page', () => {
   const expectedAboutInfo =
     'Add a trusted application to improve performance or alleviate conflicts with other applications running on your hosts. Trusted applications will be applied to hosts running Endpoint Security.';
 
+  let mockedContext: AppContextTestRender;
   let history: AppContextTestRender['history'];
   let coreStart: AppContextTestRender['coreStart'];
   let waitForAction: MiddlewareActionSpyHelper['waitForAction'];
   let render: () => ReturnType<AppContextTestRender['render']>;
   const originalScrollTo = window.scrollTo;
+  const act = reactTestingLibrary.act;
+
+  const getFakeTrustedApp = (): TrustedApp => ({
+    id: '1111-2222-3333-4444',
+    name: 'one app',
+    os: OperatingSystem.WINDOWS,
+    created_at: '2021-01-04T13:55:00.561Z',
+    created_by: 'me',
+    description: 'a good one',
+    entries: [
+      {
+        field: ConditionEntryField.PATH,
+        value: 'one/two',
+        operator: 'included',
+        type: 'match',
+      },
+    ],
+  });
+
+  const mockListApis = (http: AppContextTestRender['coreStart']['http']) => {
+    const currentGetHandler = http.get.getMockImplementation();
+
+    http.get.mockImplementation(async (...args) => {
+      const path = (args[0] as unknown) as string;
+      // @ts-ignore
+      const httpOptions = args[1] as HttpFetchOptions;
+
+      if (path === TRUSTED_APPS_LIST_API) {
+        return {
+          data: [getFakeTrustedApp()],
+          total: 50, // << Should be a value large enough to fulfill two pages
```

Comment:
first, :fire: for the comment explaining the number chosen.

second, 

```ts
const per_page = httpOptions?.query?.per_page ?? 20;

return {
  //...
  total: per_page*2, // need to fill two full pages
  per_page: per_page,
}
```
perhaps? but purely subjective